.github/workflows/src/summarize-checks/labelling.js

```diff
@@ -158,6 +393,366 @@ export const breakingChangesCheckType = {
   },
 };
 
+// #endregion constants
+// #region Required Labels
+
+
+/**
+ * @param {import("./labelling.js").LabelContext} context
+ * @param {string[]} existingLabels
+ * @returns {void}
+ */
+export function processArmReviewLabels(
+  context,
+  existingLabels
+) {
+  // the important part about how this will work depends how the users use it
+  // EG: if they add the "ARMSignedOff" label, we will remove the "ARMChangesRequested" and "WaitForARMFeedback" labels.
+  // if they add the "ARMChangesRequested" label, we will remove the "WaitForARMFeedback" label.
+  // if they remove the "ARMChangesRequested" label, we will add the "WaitForARMFeedback" label.
+  // so if the user or ARM team actually unlabels `ARMChangesRequested`, then we're actually ok
+  // if we are signed off, we should remove the "ARMChangesRequested" and "WaitForARMFeedback" labels
+  if (containsAll(existingLabels, ["ARMSignedOff"])) {
+    if (existingLabels.includes("ARMChangesRequested")) {
+      context.toAdd.add("ARMChangesRequested");
+    }
+    if (existingLabels.includes("WaitForARMFeedback")) {
+      context.toRemove.add("WaitForARMFeedback");
+    }
+  }
+  // if there are ARM changes requested, we should remove the "WaitForARMFeedback" label as the presence indicates that ARM has reviewed
+  else if (containsAll(existingLabels, ["ARMChangesRequested"]) && containsNone(existingLabels, ["ARMSignedOff"])) {
+    if (existingLabels.includes("WaitForARMFeedback")) {
+      context.toRemove.add("WaitForARMFeedback");
+    }
+  }
+  // finally, if ARMChangesRequested are not present, and we've gotten here by lac;k of signoff, we should add the "WaitForARMFeedback" label
+  else if (containsNone(existingLabels, ["ARMChangesRequested"])) {
+    if (!existingLabels.includes("WaitForARMFeedback")) {
+      context.toAdd.add("WaitForARMFeedback");
+    }
+  }
+}
+
+/**
+ *
+ * @param {any[]} arr
+ * @param {any[]} values
+ * @returns
+ */
+function containsAll(arr, values) {
+  return values.every((value) => arr.includes(value));
+}
+
+/**
+ *
+ * @param {any[]} arr
+ * @param {any[]} values
+ * @returns
+ */
+function containsNone(arr, values) {
```

Comment:
Rename to `includesNone`, add generic type params, and move to `.github/shared/array.js`.